plugin-gradle/src/test/java/com/diffplug/gradle/spotless/GradleIntegrationTest.java

```diff
@@ -36,6 +37,11 @@
 import com.diffplug.spotless.ResourceHarness;
 
 public class GradleIntegrationTest extends ResourceHarness {
+	@Before
+	public void gitAttributes() throws IOException {
+		write(".gitattributes", "* text eol=lf");
+	}
```

Comment:
Hmm, I'm not so clear on what writing this `.gitattributes` file does. What do the other tests do with it?

Reply:
Default line ending behavior is to match git.  We have a [`.gitattributes` file](https://github.com/diffplug/spotless/blob/master/.gitattributes) so that spotless will checkout with `\n` on every platform.

Our GradleIntegrationTests create a folder in a temp folder, and run a gradle build there.  Because those test folders don't have a `.gitattributes` file, git (on windows) will default to `\r\n`.  So now if you read a test file from the spotless test resources, and compare it to a build result, the line endings won't match.  We used to fix this in ResourceHarness by switching line endings to UNIX, which was dangerous.  Now we do it buy sticking this `.gitattributes` file into the test directory.

Reply:
Ah okay, that makes sense to me now. Thanks for clarifying things for me!

In that case, this whole PR LGTM. :+1: